pkg/api/aim/encoding/decoder.go

```diff
@@ -43,6 +43,7 @@ func (d *reader) readField() ([]byte, error) {
 }
 
 // Decode decodes input stream of data into map[string]interface{}.
+// nolint:gocyclo
```

Comment:
we should probably leave a TODO on the `gocyclo` ones to properly fix them — the linter is right that many of them are way too complex

Reply:
yes yes. I've decided to also added this moment. I have other portion of changes where you will see such a TODO's